### PR TITLE
clean resource before apply new

### LINKED ISSUE #254: Error after upgrading the latest version registration-operator
After I upgrading the registration-operator using the latest code(commit sha: 803bfb374805414222e2aff669b73588dffe3537), there are errors showing in the registration-operator's log:
```
E0628 04:39:13.727029       1 base_controller.go:270] "KlusterletController" controller failed to sync "klusterlet", err: "klusterlet/managed/klusterlet-work-clusterrolebinding.yaml" (string): ClusterRoleBinding.rbac.authorization.k8s.io "open-cluster-management:klusterlet-work:agent" is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:"rbac.authorization.k8s.io", Kind:"ClusterRole", Name:"open-cluster-management:klusterlet-work:agent"}: cannot change roleRef
```
And the status of the klusterlet showing:
```
  - lastTransitionTime: "2022-06-28T04:22:39Z"
    message: '"klusterlet/managed/klusterlet-work-clusterrolebinding.yaml" (string):
      ClusterRoleBinding.rbac.authorization.k8s.io "open-cluster-management:klusterlet-work:agent"
      is invalid: roleRef: Invalid value: rbac.RoleRef{APIGroup:"rbac.authorization.k8s.io",
      Kind:"ClusterRole", Name:"open-cluster-management:klusterlet-work:agent"}: cannot
      change roleRef'
    reason: KlusterletApplyFailed
    status: "False"
    type: Applied
```
cc @haoqing0110

### PR DESCRIPTION
**Pull request by haoqing0110**
Fix: https://github.com/open-cluster-management-io/registration-operator/issues/254
Related PRs: 
https://github.com/open-cluster-management-io/registration-operator/pull/250
https://github.com/open-cluster-management-io/registration-operator/pull/252
https://github.com/open-cluster-management-io/registration-operator/pull/234

Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 1]
```diff
@@ -331,6 +331,12 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		}
 	}
 
+	// Cleanup some resources before apply
+	err = n.cleanBeforeApply(ctx, &config)
```
Comment (qiujian16) [inline pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: is this temp for upgrade only in this release? I would add some doc here. Also maybe you should define an array on what should be removed and call delete directly?

Comment (haoqing0110) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: fixed

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**